### ISSUE TITLE
Replacement spatial layers

### ISSUE THREAD

**Issue by CEJohnston** [nbnuk/nbnatlas-issues#758]
Seven layers require updating.  These arise from the set being monitored by the ADH.  I will provide details as the pre-upload checks are completed.

**Comment by CEJohnston** [comment]
Here are the first four layers to be replaced:

1. Layer to be replaced:  ID 120
Layer display name:  Special Areas of Conservation (Scotland)
Zip file name: SAC_SCOTLAND_multipartWGS84.zip
Shapefile name: SAC_SCOTLAND_multipartWGS84.shp
Unique ID field: PA_CODE
Name field: NAME
Publication date: 01/07/2019
Topology check:  crashed during check for overlaps; otherwise OK

2. Layer to be replaced:  ID 119
Layer display name:  Special Protection Areas (Scotland)
Zip file name: SPA_SCOTLAND_multipartWGS84.zip
Shapefile name: SPA_SCOTLAND_multipartWGS84.shp
Unique ID field:   PA_CODE
Name field:  NAME
Publication date:  01/07/2019
Topology check:  crashed during check for overlaps; otherwise OK

3. Layer to be replaced:  ID 124
Layer display name:  Nature Conservation Marine Protected Areas (Scotland)
Zip file name: MPA_SCOTLAND_multipartWGS84.zip
Shapefile name: MPA_SCOTLAND_multipartWGS84.shp
Unique ID field:   PA_CODE
Name field:  NAME
Publication date:  20/05/2019
Topology check: file passed

4. Layer to be replaced:  ID 118
Layer display name:  Sites of Special Scientific Interest (Scotland)
Zip file name: SSSI_SCOTLAND_multipartWGS84.zip
Shapefile name: SSSI_SCOTLAND_multipartWGS84.shp
Unique ID field: PA_CODE
Name field: NAME
Publication date:  01/07/2019
Topology check: overlaps detected but otherwise OK.

All the above layers have been worked on to create multi part polygons and to convert to WGS84.  Where topology checks reveal overlapping polygons I am not able to fix them as I need a additional plugin to do the work - if you could correct them for input that would be useful.  Still having problems with QGIS crashing during checks.

**Comment by CEJohnston** [comment]
Here are the details of the final 3 layers for this upload:

5.  Layer to be replaced:  ID 47
Layer display name:  Conservation Areas (Scotland)
Zip file name: Conservation_AreasWGS84.zip
Shapefile name: Conservation_AreasWGS84.shp
Unique ID field:  DES_TITLE
Name field: DES_REF 
Publication date:  19/02/2019
Topology checks:  38 overlaps; 8 invalid geometries reported; duplicates fixed

6.  Layer to be replaced:  ID 33
Layer display name:  Landscape Character Assessment (Scotland)
Zip file name: LCA_SCOTLAND_WGS84.zip
Shapefile name: LCA_SCOTLAND_WGS84.shp
Unique ID field: LCT_NO
Name field: LCT
Publication date:  01/07/2019
Topology checks:  12 overlaps detected; duplicates fixed.

7.  Layer to be replaced:  ID 37
Layer display name:  Nitrate Vulnerable Zones (Scotland)
Zip file name: Scotland_NVZones_2016_WGS84.zip
Shapefile name: Scotland_NVZones_2016_WGS84.shp
Unique ID field:  OBJECTID
Name field: NVZName
Publication date:  16/11/2016
Topology check: passed

All these layers have been converted to WGS84 and have been uploaded to the folder on Teams.

**Comment by sophiathirza** [comment]
I think that this mean that only 3 and 7 can be uploaded as is.

**Comment by CEJohnston** [comment]
Here is the link to the Teams folder containing the seven files: [https://teams.microsoft.com/_#/files/Spatial%20layer%20upload?threadId=19%3A3d6afc80fde447a18b26047d950d5ac9%40thread.skype&ctx=channel](url) 
As discussed could all seven be tested - am hoping the portal can tolerate them as they are just replacements.  Thanks.

**Comment by reupost** [comment]
@CEJohnston 
@sophiathirza 
All layers loaded successfully. I've checked and they display on the spatial portals. Please could you
(a) confirm that they look ok
(b) add metadata as needed
Then I'll disable the old layers and ensure that any references to the old layers are replaced with the new layer IDs.

**Comment by CEJohnston** [comment]
I have now checked the above layers.  Replacement layers 1-4 and 6-7 (as numbered in the above list) look and sample OK.  Their metadata has been completed.  The old versions of these layers can now be disabled.

For replacement layer 5 (above) unfortunately I made an error in the instructions for the fields to be used for the upload of the new layer - I reversed the attributes.  Please remove the new layer (ID 172) completely and re-upload the replacement.  

The **corrected** upload details are: 

Layer to be replaced: ID 47
Layer display name: Conservation Areas (Scotland)
Zip file name: Conservation_AreasWGS84.zip
Shapefile name: Conservation_AreasWGS84.shp
**Unique ID field: DES_REF
Name field: DES_TITLE**
Publication date: 19/02/2019

I’ll add the metadata to the new version after upload.  Thanks very much for uploading these replacements even though they failed the topology checks.  It does look like the portal can tolerate some imperfect shapefiles.

**Comment by CEJohnston** [comment]
I have checked the correct version of the Conservation Areas (Scotland) layer (ID 186) which was uploaded in November data processing run and it is OK.  Metadata has now been updated.

The old layers that have been replaced are still to be disabled so that they no longer appear in the layer chooser and any references to the old layers are replaced with the new layer IDs. 

For reference, the OLD layer IDs are:
120 -  Special Areas of Conservation (Scotland) 
119 - Special Protection Areas (Scotland)
124 - Nature Conservation Marine Protected Areas (Scotland)
33 - Landscape Character Assessment (Scotland)
37- Nitrate Vulnerable Zones (Scotland)
47 - Conservation Areas (Scotland)
118 - Sites of Special Scientific Interest (Scotland) (I think this is already disabled)

For reference, the NEW layer IDs are:
177 -  Special Areas of Conservation (Scotland) *
176 - Special Protection Areas (Scotland) *
175 - Nature Conservation Marine Protected Areas (Scotland) *
174 - Landscape Character Assessment (Scotland)
171 - Nitrate Vulnerable Zones (Scotland) *
186 - Conservation Areas (Scotland)
173 - Sites of Special Scientific Interest (Scotland) *

Note: the layers marked * are used in the NBN Atlas Scotland in the 'Select a region to explore' in the Regions portal.

I have made a new Trello card for this work so that this GitHub issue can be closed when work done.  
The card is: https://trello.com/c/GW8mYAT9/263-spatial-layers-outstanding-work-following-the-november-processing-run
